static/app/views/performance/landing/widgets/transforms/transformMetricsToArea.tsx

```diff
@@ -20,58 +22,146 @@ export function transformMetricsToArea<T extends WidgetDataConstraint>(
 
   const metricsField = widgetProps.fields[0];
 
-  const data =
-    response?.groups.map(group => {
+  const commonChildData = {
+    loading,
+    reloading,
+    isLoading: loading || reloading,
+    isErrored: errored,
+
+    utc: utc === 'true',
+    interval,
+    statsPeriod: statsPeriod ?? undefined,
+    start: start ?? '',
+    end: end ?? '',
+  };
+
+  if (!response || !responsePrevious) {
+    return {
+      ...commonChildData,
+      hasData: false,
+      data: [],
+      dataMean: undefined,
+      previousData: undefined,
+    };
+  }
+
+  if (widgetProps.chartSetting === PerformanceWidgetSetting.FAILURE_RATE_AREA) {
```

Comment:
There seems to be a lot of repetition with "this" and "previous" failure rate. Any chance we could DRY it up a bit?

Reply:
yes I noticed that too... I will think about something :)